parachain/runtime/litentry/src/xcm_config.rs

```diff
@@ -82,152 +68,103 @@ pub type LocationToAccountId = (
-	// ^^^ Parent and its exec plurality get free execution
-);
+	// Parent & its unit plurality gets free execution.
+	AllowExplicitUnpaidExecutionFrom<ParentOrParentsPlurality>,
+	// Expected responses are OK.
+	AllowKnownQueryResponses<PolkadotXcm>,
+	// Subscriptions for version tracking are OK.
+	AllowSubscriptionsFrom<Everything>,
+	// HRMP notifications from the relay chain are OK.
+	AllowHrmpNotificationsFromRelayChain,
+)>;
 
 parameter_types! {
-	/// Xcm fees will go to the treasury account
-	pub XcmFeesAccount: AccountId = Treasury::account_id();
+	// One XCM operation is 1_000_000_000 weight - almost certainly a conservative estimate.
+	pub UnitWeightCost: Weight = Weight::from_parts(1_000_000_000, 64 * 1024);
 	pub const MaxAssetsIntoHolding: u32 = 64;
-	pub const WeightToFeeFactor: Balance = WEIGHT_TO_FEE_FACTOR; // 10^6
+	pub const MaxInstructions: u32 = 100;
 }
 
-pub type Traders = (
-	UsingComponents<
-		ConstantMultiplier<Balance, WeightToFeeFactor>,
-		NewAnchoringSelfReserve<Runtime>,
-		AccountId,
-		Balances,
-		DealWithFees<Runtime>,
-	>,
-	UsingComponents<
-		ConstantMultiplier<Balance, WeightToFeeFactor>,
-		OldAnchoringSelfReserve<Runtime>,
-		AccountId,
-		Balances,
-		DealWithFees<Runtime>,
-	>,
-	// TODO::Implement foreign asset fee to weight rule from AssetManager Setting; Need more test
-	FirstAssetTrader<
-		CurrencyId<Runtime>,
-		AssetManager,
-		XcmFeesToAccount<
-			Assets,
-			ConvertedConcreteId<AssetId, Balance, AssetIdMultiLocationConvert<Runtime>, JustTry>,
-			AccountId,
-			XcmFeesAccount,
-		>,
-	>,
-);
-
-/// Xcm Weigher shared between multiple Xcm-related configs.
-pub type XcmWeigher = FixedWeightBounds<UnitWeightCost, RuntimeCall, MaxInstructions>;
-
 pub struct XcmConfig;
 impl xcm_executor::Config for XcmConfig {
 	type RuntimeCall = RuntimeCall;
 	type XcmSender = XcmRouter;
-	// How to withdraw and deposit an asset.
 	type AssetTransactor = AssetTransactors;
 	type OriginConverter = XcmOriginToTransactDispatchOrigin;
-	// Only Allow chains to handle their own reserve assets crossed on local chain whatever way they
-	// want.
-	type IsReserve = MultiNativeAsset;
-	type IsTeleporter = (); // Teleporting is disabled.
+	type IsReserve = NativeAsset; // TODO
+	type IsTeleporter = ();
 	type UniversalLocation = UniversalLocation;
-	type Barrier = Barriers;
-	type Weigher = XcmWeigher;
-	// Litentry: This is the tool used for calculating that inside XcmExecutor vm, how to transfer
-	// asset into weight fee. Usually this is in order to fulfull Barrier
-	// AllowTopLevelPaidExecutionFrom requirement. Currently we have not implement the asset to fee
-	// rule for Foreign Asset, so pure cross chain transfer from XCM parachain will be rejected no
-	// matter.
-	type Trader = Traders;
+	type Barrier = Barrier;
+	type Weigher = FixedWeightBounds<UnitWeightCost, RuntimeCall, MaxInstructions>;
+	type Trader = UsingComponents<IdentityFee<Balance>, SelfLocation, AccountId, Balances, ()>;
```

Comment:
This comment is just for recording purpose.

So this impl will make us support native token only for now.

Reply:
Yes this is the goal (for now)